src/libstd/macros.rs

```diff
@@ -210,21 +217,15 @@ macro_rules! debug_assert_eq {
 /// }
 /// ```
 #[macro_export]
+#[stable]
 macro_rules! unreachable {
-    () => ({
-        panic!("internal error: entered unreachable code")
-    });
-    ($msg:expr) => ({
-        unreachable!("{}", $msg)
-    });
-    ($fmt:expr, $($arg:tt)*) => ({
-        panic!(concat!("internal error: entered unreachable code: ", $fmt), $($arg)*)
-    });
+    () => (panic!("internal error: entered unreachable code"))
```

Comment:
What's the rationale for dropping the functionality here, rather than beefing up `unimplemented`?